lib/std/fs.zig

```diff
@@ -2190,10 +2190,18 @@ pub fn selfExePathAlloc(allocator: *Allocator) ![]u8 {
 /// TODO make the return type of this a null terminated pointer
 pub fn selfExePath(out_buffer: []u8) SelfExePathError![]u8 {
     if (is_darwin) {
-        var u32_len: u32 = @intCast(u32, math.min(out_buffer.len, math.maxInt(u32)));
-        const rc = std.c._NSGetExecutablePath(out_buffer.ptr, &u32_len);
+        // Note that _NSGetExecutablePath() will return "a path" to
+        // the executable not a "real path" to the executable.
+        var symlink_path_buf: [MAX_PATH_BYTES]u8 = undefined;
+        var u32_len: u32 = MAX_PATH_BYTES;
+        const rc = std.c._NSGetExecutablePath(&symlink_path_buf, &u32_len);
```

Comment:
Since `_NSGetExecutablePath` first parameter is null-terminated and this is the only callsite can you change the signature in `darwin.zig`? 
That'd also help to get rid of the `ptrCast` below